src/sync/stream.ts

```diff
@@ -51,6 +65,114 @@ function isStreamSyncHistoryError(data: IStreamSyncHistoryData): data is IStream
   return data.type === 'stream:history:error';
 }
 
+interface IStreamItemVertex {
+  seq: number;
+  type: 'vertex';
+  vertex: IHistoryTx;
+}
+
+interface IStreamItemAddress {
+  seq: number;
+  type: 'address';
+  address: IAddressInfo;
+}
+
+type IStreamItem = IStreamItemAddress | IStreamItemVertex;
+
+function isStreamItemVertex(item: IStreamItem): item is IStreamItemVertex {
+  return item.type === 'vertex';
+}
+
+function isStreamItemAddress(item: IStreamItem): item is IStreamItemAddress {
+  return item.type === 'address';
+}
+
+class StreamStatsManager {
+  recvCounter: number;
+
+  procCounter: number;
+
+  ackCounter: number;
+
+  emptyCounter: number;
+
+  inTimer?: ReturnType<typeof setTimeout>;
+
+  outTimer?: ReturnType<typeof setTimeout>;
+
+  qTimer?: ReturnType<typeof setTimeout>;
+
+  logger: ILogger;
+
+  q: Queue;
+
+  dt: number;
+
+  constructor(queue: Queue, logger: ILogger) {
+    this.recvCounter = 0;
+    this.procCounter = 0;
+    this.ackCounter = 0;
+    this.emptyCounter = 0;
+
+    this.logger = logger;
+    this.dt = 2000;
+    this.q = queue;
+    this.qTimer = setInterval(() => {
+      this.logger.debug(`[*] queue_size: ${queue.size()}`);
+    }, this.dt);
+  }
+
+  clean() {
+    if (this.inTimer) {
+      clearTimeout(this.inTimer);
+    }
+    if (this.outTimer) {
+      clearTimeout(this.outTimer);
+    }
+    if (this.qTimer) {
+      clearInterval(this.qTimer);
+    }
+  }
+
+  ack(seq: number) {
+    this.ackCounter += 1;
+    this.logger.debug(
+      `[*] ACKed ${seq} with queue at ${this.q.size()}. Sent ACK ${this.ackCounter} times`
+    );
+  }
+
+  recv() {
+    if (!this.inTimer) {
+      this.inTimer = setTimeout(() => {
```

Comment:
For now this functionality only gives us some logs for statistics purpuses, testing logs seems inneficient.
In the future this will be using this for window maintenance and the tests will be quite relevant, but not now.